extensions/sql-migration/src/dialog/configureIR/configureIRDialog.ts

```diff
@@ -0,0 +1,509 @@
+
+		const service = this._migrationStateModel._sqlMigrationService;
+		if (service) {
+			const account = this._migrationStateModel._azureAccount;
+			const subscription = this._migrationStateModel._sqlMigrationServiceSubscription;
+			const resourceGroup = service.properties.resourceGroup;
+			const location = service.location;
+			const serviceName = service.name;
+			if (service?.properties?.integrationRuntimeState) {
+				service.properties.integrationRuntimeState = undefined;
+			}
+
+			const migrationService = await getSqlMigrationService(
+				account,
+				subscription,
+				resourceGroup,
+				location,
+				serviceName);
+
+			await refreshAuthenticationKeyTable(view,
+				this._migrationServiceAuthKeyTable,
+				account,
+				subscription,
+				resourceGroup,
+				location,
+				migrationService);
+
+		}
+
+		container.addItems([instructions, this._migrationServiceAuthKeyTable]);
+
+		return container;
+	}
+
+	// creates powershell script container
+	private async createPowershellscriptContentContainer(view: azdata.ModelView): Promise<azdata.Component> {
+		const container = view.modelBuilder.flexContainer().withLayout({
+			flexFlow: 'column',
+		}).component();
+
+		const def = view.modelBuilder.text().withProps({
+			value: constants.POWERSHELL_SCRIPT_DESCRIPTION,
+			CSSStyles: {
+				'font-size': '13px',
+				'line-height': '18px',
+				'margin': '0px',
+				'margin-top': '8px',
+				'margin-bottom': '8px',
+			},
+		}).component();
+
+		const saveScriptButton = view.modelBuilder.button().withProps({
+			label: 'Save script',
+			ariaLabel: 'Save script',
+			iconWidth: '18px',
+			iconHeight: '18px',
+			height: '40px',
+			width: '100px',
+			iconPath: IconPathHelper.save,
+			CSSStyles: {
+				'padding': '8px 8px',
+			},
+		}).component();
+
+		const scriptContent = await fs.readFile(path.join(__dirname, '../../../scripts/SHIR-auto-configuration.ps1'));
+
+		// inject auth keys in the script
+		const authKeys = await this.retrieveAuthKeys();
+		const modifiedScriptContent = await this.injectKeysIntoShirScriptContent
+			(authKeys.authKey1, authKeys.authKey2, scriptContent.toString());
+
+		// write it back to different file
+		const modifiedScriptPath = path.join(__dirname, '../../../scripts/SHIR-auto-configuration-with-auth-keys.ps1');
+		await fs.writeFile(modifiedScriptPath, modifiedScriptContent);
+
+		saveScriptButton.onDidClick(async () => {
+			const options: vscode.SaveDialogOptions = {
+				defaultUri: vscode.Uri.file(suggestReportFile(Date.now())),
+				filters: { 'Windows PowerShell Script': ['ps1'] }
+			};
+
+			const choosenPath = await vscode.window.showSaveDialog(options);
+			if (choosenPath !== undefined) {
+				const value = modifiedScriptContent.toString();
+				await fs.writeFile(choosenPath.fsPath, value);
+				if (await vscode.window.showInformationMessage(
+					"PowerShell script saved",
+					"Open", "Cancel") === "Open") {
```

Comment:
need to create resource strings for these as well for localization.

Reply:
Done